pypeit/core/findobj_skymask.py

```diff
@@ -993,286 +1081,181 @@ def objs_in_slit(image, thismask, slit_left, slit_righ, inmask=None, fwhm=3.0, u
-        # Trim to only objects above this threshold
-        ikeep = (ypeak >= threshold)
-        xcen = xcen[ikeep]
-        ypeak = ypeak[ikeep]
-        nobj_reg = len(xcen)
-        # Now create SpecObj objects for all of these
-        for iobj in range(nobj_reg):
-            thisobj = specobj.SpecObj(**specobj_dict)
-            #
-            thisobj.SPAT_FRACPOS = xcen[iobj]/nsamp
-            thisobj.smash_peakflux = ypeak[iobj]
-            thisobj.smash_nsig = ypeak[iobj]/sigma
-            sobjs.add_sobj(thisobj)
-    else:
-        nobj_reg = 0
-
-    # ToDo Also plot the edge trimming boundaries on the QA here.
-    if show_peaks or objfindQA_filename is not None:
-        spat_approx_vec = slit_left[specmid] + xsize[specmid]*np.arange(nsamp)/nsamp
-        spat_approx = slit_left[specmid] + xsize[specmid]*xcen/nsamp
-        # Define the plotting function
-        #plt.plot(spat_approx_vec, fluxsub/sigma, color ='cornflowerblue',linestyle=':', label='Collapsed Flux')
-        plt.plot(spat_approx_vec, fluxconv_cont/sigma, color='black', label = 'Collapsed flux (FWHM convol)')
-        plt.plot(spat_approx_vec[cont_mask], fluxconv_cont[cont_mask]/sigma, color='red', markersize=3.0,
-                 mfc='red', linestyle='None', fillstyle='full',
-                 zorder=9, marker='o', label = 'Used for threshold')
-        plt.hlines(threshold/sigma,spat_approx_vec.min(),spat_approx_vec.max(), color='red',linestyle='--', label='Threshold')
-        plt.hlines(1.0,spat_approx_vec.min(),spat_approx_vec.max(), color='green',linestyle=':', label='+- 1 sigma')
-        plt.hlines(-1.0,spat_approx_vec.min(),spat_approx_vec.max(), color='green',linestyle=':')
-
-        plt.plot(spat_approx, ypeak/sigma, color='red', marker='o', markersize=10.0, mfc='lawngreen', fillstyle='full',
-        linestyle='None', zorder = 10,label='Object Found')
-        plt.legend()
-        plt.xlabel('Approximate Spatial Position (pixels)')
-        plt.ylabel('F/sigma (significance)')
-        # plt.title(qa_title + ': Slit# {:d}'.format(objfindQA_dict['SLITORD_ID']))
-        plt.title(qa_title)
-        if objfindQA_filename is not None:
-            plt.savefig(objfindQA_filename, dpi=400)
-        if show_peaks:
-            viewer, ch = display.show_image(image*(thismask*inmask))
-            plt.show()
-        plt.close('all')
-
-    # Now loop over all the regular apertures and assign preliminary traces to them.
+    # Trim to the good peaks
+    x_peaks = x_peaks_all[peaks_gpm]
+    snr_peaks = snr_peaks_all[peaks_gpm]
+    flux_peaks = flux_peaks_all[peaks_gpm]
+
+    # Now create SpecObj objects for all of these and assign preliminary traces to them.
+    for iobj in range(nobj_reg):
+        thisobj = specobj.SpecObj(**specobj_dict)
+        thisobj.SPAT_FRACPOS = x_peaks[iobj]/nsamp
+        thisobj.smash_peakflux = flux_peaks[iobj]
+        thisobj.smash_snr = snr_peaks[iobj]
+        sobjs.add_sobj(thisobj)
```

Comment:
Add a call to `ready_for_extraction` here

Reply:
Or wherever you are done tinkering with the SpecObjs

Reply:
Well that was not being done in the previous version as far as I can tell so I'm reluctant to add it now. There is a call to this at the end of the routine which is where I think it probably ought to be.